### PR TITLE
Custom reports continued -- rebase of 2096

### PR DESCRIPTION
**Pull request by quoideneuf**
See https://github.com/archivesspace/archivesspace/pull/2096

and

JIRA
ANW-1072
ANW-1076
ANW-1073
ANW-1184
ANW-1068
ANW-1254

custom reports back in
get rid of unneccesary code

make custom reports a configurable option

Add limit options to custom report template

Rename db migration with new number

ANW-1072 Fix archival objects subreport for resources

Add check all and uncheck all functionality to create and edit of custom report templates

ANW-1073 Add copy of custom template

Rebase master - rename migration

[ANW-1073] Fix bug

[ANW-1184] Remove language as a custom field for resources, archival objects, digital objects and digital object components
and add language as a sub record

ANW-1076 check/uncheck for display checkboxes only (#2154)

ANW 1073 and ANW 1076 Improvements (#2161)

* ANW-1076 check/uncheck for display checkboxes only

* ANW-1073 improvements

Additional button enhancements (#2162)

Update mimemagic version (#2167)

Rename migration for rebase to master (#2175)

Update colors and placement of buttons (#2180)

ANW-1184 add missing translation (#2184)

[ANW-1068] Add Create Template button to custom report background job (#2188)

[ANW-1254] Add breadcrumb back to Custom Reports browse page from Create New Custom Report Template page (#2198)
And run rubocop

Add Copy to non-en locales files (#2199)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have authority to submit this code.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

[ANW-1073]: https://archivesspace.atlassian.net/browse/ANW-1073
[ANW-1184]: https://archivesspace.atlassian.net/browse/ANW-1184
[ANW-1068]: https://archivesspace.atlassian.net/browse/ANW-1068

### CODE REVIEW

**Comment by cdibella** [comment]
@lorawoodford @mark-cooper It's been a long road, as we know, but all code has been submitted for this now. Please review it (from a code standpoint) as soon as you can so we can get it off a branch and/or @lmcglohon can fix the code problems while it's still fresh in her mind. Thank you!

File: frontend/app/views/custom_report_templates/_form.html.erb [hunk 1]
```diff
@@ -10,10 +10,12 @@
 	<% record_types.delete('global') %>
 
 	<div id="template_information">
-		<h3><%= I18n.t "custom_report_template._frontend.section.template_information" %></h3>
-	
+		<h3><%= I18n.t "custom_report_template._frontend.section.template_information" %>
+		<%= link_to_help :topic => "accession_basic_information" %></h3>
```
Comment (lorawoodford) [inline frontend/app/views/custom_report_templates/_form.html.erb:14]: Is there a valid help topic to link this to now?  It def should be going to accession basic info.

Comment (cdibella) [reply frontend/app/views/custom_report_templates/_form.html.erb:14]: This is a placeholder just to get the question mark in there - we haven't created the manual page yet (though we will before custom reports are in a release).

File: frontend/app/views/custom_report_templates/index.html.erb [hunk 2]
```diff
@@ -7,6 +7,7 @@
 		<div class="record-toolbar">
 			<div class="btn-group pull-right">
 				<%= link_to I18n.t("custom_report_template._frontend.action.create"), {:controller => :custom_report_templates, :action => :new}, :class => "btn btn-sm btn-default" %>
+				<%= link_to_help :topic => "accession_basic_information" %>
```
Comment (lorawoodford) [inline frontend/app/views/custom_report_templates/index.html.erb:10]: Same as above... need updated help topic

File: common/locales/en.yml [hunk 3]
```diff
@@ -2910,4 +2923,25 @@ en:
     log_validate_only: "Skipping creation and linking of top containers since validate only option is enabled."
     log_skipping_due_to_errors: "Skipping creation and linking of top containers due to errors."
     log_validating: "Validating spreadsheet data..."
-    log_creating_and_linking: "Creating and linking top containers..."
\ No newline at end of file
+    log_creating_and_linking: "Creating and linking top containers..."
+
+  ratings:
```
Comment (lorawoodford) [inline common/locales/en.yml:2928]: Did these slip in accidentally, or do these really need to be here?

**Comment by lorawoodford** [comment]
Aside from the two simple fixes to the help text above and the one question about the inclusion of translations for assessments, this looks good to merge to me.  I have a list of (presumably low effort) functionality enhancements I'd like to recommend we implement sooner rather than later, but those can be added after this first pass is in if we want to do them.  As this stands now it works and delivers what was requested.  

Over to you @mark-cooper

File: frontend/app/views/custom_report_templates/_form.html.erb (hunk 1 above)
Comment (cdibella) [reply frontend/app/views/custom_report_templates/_form.html.erb:14]: This is a blank manual page that will eventually have documentation for custom reports: https://archivesspace.atlassian.net/wiki/spaces/ArchivesSpaceUserManual/pages/2667839513/Creating+Custom+Reports+coming+for+ArchivesSpace+3.1.0. It's currently private, but the URL will stay the same, and Confluence provides redirects if the name changes.

**Review by lorawoodford — DISMISSED** [review]

**Review by lorawoodford — APPROVED** [review]

**Pull request merged**